src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -157,46 +152,70 @@ public SGAAssemblyResult(final File contigFile, final List<Tuple2<String, Tuple3
-     *                              or when copying the FASTQ file from HDFS to local disk.
      */
     @VisibleForTesting
-    static Tuple2<Long, SGAAssemblyResult> performAssembly(final Tuple2<Long, URI> fastqOfABreakpoint,
-                                                           final Path sgaPath,
-                                                           final int threads,
-                                                           final boolean runCorrections)
-            throws IOException{
-
-        final Long breakpointId = fastqOfABreakpoint._1();
-        final String pathStringToRawFASTQ = fastqOfABreakpoint._2().getPath();
-        final String rawFASTQFileName = FilenameUtils.getName(pathStringToRawFASTQ);
-
-        final File workingDir = Files.createTempDirectory( "assembly" + breakpointId.toString() ).toAbsolutePath().toFile();
-        workingDir.deleteOnExit();
+    static Tuple2<Long, PipeLineResult> performAssembly(final FileSystem fs,
+                                                        final Tuple2<Long, URI> fastqOfABreakpoint,
+                                                        final Path sgaPath,
+                                                        final int threads,
+                                                        final boolean runCorrections){
+
+        final Long breakpointID = fastqOfABreakpoint._1();
+        final URI  uriToFASTQ   = fastqOfABreakpoint._2();
+        if(null==uriToFASTQ){
+            return new Tuple2<>(breakpointID, new PipeLineResult("Failed to parse URI for breakpoint: " + breakpointID.toString()));
+        }
 
-        BucketUtils.copyFile(pathStringToRawFASTQ, null, workingDir.toPath().toAbsolutePath().toString()+"/"+rawFASTQFileName);
+        final Tuple2<File, String> copySuccess = makeTempDirAndCopyFASTQToLocal(fs, uriToFASTQ, breakpointID);
+        final File workingDir = copySuccess._1();
+        if(workingDir!=null){
+            final PipeLineResult assembledContigsFileAndRuntimeInfo = runSGAModulesInSerial(sgaPath, workingDir, copySuccess._2(), threads, runCorrections);
+            return new Tuple2<>(breakpointID, assembledContigsFileAndRuntimeInfo);
+        }else{
+            return new Tuple2<>(breakpointID, new PipeLineResult(copySuccess._2()));
+        }
+    }
 
-        final SGAAssemblyResult assembledContigsFileAndRuntimeInfo = runSGAModulesInSerial(sgaPath, workingDir, rawFASTQFileName, threads, runCorrections);
+    /**
+     * Copy from hdfs to temp local dir
+     * @param fs            filesystem
+     * @param uriToFASTQ    uri to source
+     * @param breakpointID  breakpoint ID
+     * @return              (temp directory, FASTQ file name) if successfully copied, or (null, error message) if failed
+     */
+    @VisibleForTesting
+    static Tuple2<File, String> makeTempDirAndCopyFASTQToLocal(final FileSystem fs, final URI uriToFASTQ, final Long breakpointID){
```

Comment:
Rather than return a Tuple2 of the directory and filename, why not just always name the fastq the same thing and just return the directory?